pytest/lib/utils.py

```diff
@@ -190,24 +196,27 @@ def load_test_contract(filename='test_contract_rs.wasm'):
 
 
 def compile_rust_contract(content):
-    empty_contract_rs = os.path.join(os.path.dirname(__file__),
-                                     '../empty-contract-rs')
-    run('mkdir -p /tmp/near')
-    tmp_contract = tempfile.TemporaryDirectory(dir='/tmp/near').name
-    p = run(f'cp -r {empty_contract_rs} {tmp_contract}')
-    if p.returncode != 0:
-        raise Exception(p.stderr)
-
-    with open(f'{tmp_contract}/src/lib.rs', 'a') as f:
-        f.write(content)
-
-    p = run('bash', input=f'''
-cd {tmp_contract}
-./build.sh
-''')
-    if p.returncode != 0:
-        raise Exception(p.stderr)
-    return f'{tmp_contract}/target/release/empty_contract_rs.wasm'
+    tempdir = pathlib.Path(tempfile.gettempdir()) / 'near'
+    tempdir.mkdir(parents=True, exist_ok=True)
+    with tempfile.TemporaryDirectory(dir=tempdir) as build_dir:
+        build_dir = pathlib.Path(build_dir) / 'empty-contract-rs'
+        shutil.copytree(
+            pathlib.Path(__file__).parent.parent / 'empty-contract-rs',
+            build_dir, symlinks=True)
+        (build_dir / 'src').mkdir(parents=True, exist_ok=True)
+        with open(build_dir / 'src' / 'lib.rs', 'w') as wr:
+            wr.write(content)
+        subprocess.check_call(('cargo', 'build', '--release', '--target',
+                               'wasm32-unknown-unknown'), cwd=build_dir)
+        wasm_src = (build_dir / 'target' / 'wasm32-unknown-unknown' /
+                    'release' / 'empty_contract_rs.wasm')
+        # Yes, we are leaving this file in /tmp/near and there’s no automatic
+        # cleanup.  Yes, we probably should fix that at some point so that this
+        # temporary is deleted once the test finishes.
```

Comment:
Should we create an issue for this?

Reply:
https://github.com/near/nearcore/pull/4494